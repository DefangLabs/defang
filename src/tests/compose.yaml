name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
