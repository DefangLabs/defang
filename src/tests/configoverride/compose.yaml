services:
  service1:
    image: alpine
    environment:
      - VAR1=service1 # this is the config returned by the mock client
