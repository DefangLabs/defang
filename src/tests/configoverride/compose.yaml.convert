[
  {
    "name": "service1",
    "image": "alpine",
    "platform": 2,
    "internal": true,
    "secrets": [
      {
        "source": "VAR1"
      }
    ],
    "networks": 1
  }
]