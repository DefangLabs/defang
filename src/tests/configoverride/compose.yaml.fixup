[
  {
    "command": null,
    "entrypoint": null,
    "environment": {
      "VAR1": null
    },
    "image": "alpine",
    "networks": {
      "default": null
    }
  }
]