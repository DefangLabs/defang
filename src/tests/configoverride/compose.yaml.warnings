 ! service "service1": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "service1": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors