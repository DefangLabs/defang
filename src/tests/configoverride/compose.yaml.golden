name: configoverride
services:
  service1:
    environment:
      VAR1: service1test
    image: alpine
    networks:
      default: null
networks:
  default:
    name: configoverride_default
